modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/processor/OperationMessageProcessor.java

```diff
@@ -258,7 +264,11 @@ private MetadataContext getMetadataContext() throws MetadataResolvingException
         }
 
         ConfigurationInstance<Object> configuration = getConfiguration(getInitialiserEvent(muleContext));
-        return new DefaultMetadataContext(configuration, connectionManager);
+
+        //TODO MULE-9530: Improve id creation, this won't work with dynamic configurations
+        String cacheId =  configuration.getName();
+        MetadataCache cache = metadataManager.getCache(cacheId);
+        return new DefaultMetadataContext(configuration, connectionManager, cache);
```

Comment:
remove variable and just call the getCache? I'd leave the cacheId separated for now, until the referenced ticket is done